### PR TITLE
chore: Add migrations script to postgres AMI

### PR DESCRIPTION
**Pull request by Lakshmipathi**
Signed-off-by: Lakshmipathi <lakshmi@supabase.io>

### CODE REVIEW

**Comment by Lakshmipathi** [comment]
AMI created successfully: ami-0a737a5b2a5788afb 
Todo:
- use this image as db instance and test project creation.

**Comment by Lakshmipathi** [comment]
WIP: Testing current changes locally.

**Comment by Lakshmipathi** [comment]
Task timings:
```
    amazon-ebssurrogate.source: PLAY RECAP *********************************************************************
    amazon-ebssurrogate.source: /mnt                       : ok=318  changed=267  unreachable=0    failed=0    skipped=11   rescued=0    ignored=1   
    amazon-ebssurrogate.source:
    amazon-ebssurrogate.source: Playbook run took 0 days, 0 hours, 42 minutes, 17 seconds
    amazon-ebssurrogate.source: Wednesday 19 October 2022  07:47:26 +0000 (0:00:00.416)       0:42:17.640 *****
    amazon-ebssurrogate.source: ===============================================================================
    amazon-ebssurrogate.source: plv8 - build ---------------------------------------------------------- 723.42s
    amazon-ebssurrogate.source: Postgres - build ------------------------------------------------------ 373.35s
    amazon-ebssurrogate.source: pg_jsonschema - install ----------------------------------------------- 242.25s
    amazon-ebssurrogate.source: install cargo-pgx ----------------------------------------------------- 187.00s
    amazon-ebssurrogate.source: postgis - build SFCGAL ------------------------------------------------ 168.74s
    amazon-ebssurrogate.source: wal-g - deps ---------------------------------------------------------- 145.15s
    amazon-ebssurrogate.source: libsodium - build ------------------------------------------------------ 70.71s
    amazon-ebssurrogate.source: wal-g - build and install ---------------------------------------------- 49.98s
    amazon-ebssurrogate.source: pgRouting - build ------------------------------------------------------ 46.04s
    amazon-ebssurrogate.source: nginx - build ---------------------------------------------------------- 34.95s
    amazon-ebssurrogate.source: postgis - build -------------------------------------------------------- 24.96s
    amazon-ebssurrogate.source: timescaledb - build ---------------------------------------------------- 24.74s
    amazon-ebssurrogate.source: pljava - build --------------------------------------------------------- 21.78s
    amazon-ebssurrogate.source: Postgres - configure --------------------------------------------------- 17.52s
    amazon-ebssurrogate.source: install rust/cargo ----------------------------------------------------- 14.57s
    amazon-ebssurrogate.source: PgBouncer - build ------------------------------------------------------ 14.11s
    amazon-ebssurrogate.source: pg_graphql - build libgraphqlparser ------------------------------------ 11.10s
    amazon-ebssurrogate.source: Remove build dependencies ---------------------------------------------- 10.58s
    amazon-ebssurrogate.source: libsodium - configure --------------------------------------------------- 9.88s
    amazon-ebssurrogate.source: PgBouncer - configure --------------------------------------------------- 9.61s
    amazon-ebssurrogate.source: Wednesday 19 October 2022  07:47:26 +0000 (0:00:00.418)       0:42:17.641 *****
    amazon-ebssurrogate.source: ===============================================================================
    amazon-ebssurrogate.source: make ----------------------------------------------------------------- 1741.72s
    amazon-ebssurrogate.source: shell ----------------------------------------------------------------- 550.14s
    amazon-ebssurrogate.source: apt -------------------------------------------------------------------- 96.60s
    amazon-ebssurrogate.source: unarchive -------------------------------------------------------------- 34.74s
    amazon-ebssurrogate.source: get_url ---------------------------------------------------------------- 21.14s
    amazon-ebssurrogate.source: git -------------------------------------------------------------------- 20.12s
    amazon-ebssurrogate.source: file ------------------------------------------------------------------- 20.08s
    amazon-ebssurrogate.source: template --------------------------------------------------------------- 12.54s
    amazon-ebssurrogate.source: copy ------------------------------------------------------------------- 11.87s
    amazon-ebssurrogate.source: ufw --------------------------------------------------------------------- 9.19s
    amazon-ebssurrogate.source: systemd ----------------------------------------------------------------- 7.63s
    amazon-ebssurrogate.source: lineinfile -------------------------------------------------------------- 2.96s
    amazon-ebssurrogate.source: user -------------------------------------------------------------------- 2.90s
    amazon-ebssurrogate.source: replace ----------------------------------------------------------------- 1.68s
    amazon-ebssurrogate.source: gather_facts ------------------------------------------------------------ 1.11s
    amazon-ebssurrogate.source: find -------------------------------------------------------------------- 0.73s
    amazon-ebssurrogate.source: group ------------------------------------------------------------------- 0.72s
    amazon-ebssurrogate.source: pam_limits -------------------------------------------------------------- 0.66s
    amazon-ebssurrogate.source: community.general.snap -------------------------------------------------- 0.42s
    amazon-ebssurrogate.source: set_fact ---------------------------------------------------------------- 0.36s
    amazon-ebssurrogate.source: ansible.builtin.stat ---------------------------------------------------- 0.26s
    amazon-ebssurrogate.source: ansible.builtin.fail ---------------------------------------------------- 0.03s
    amazon-ebssurrogate.source: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    amazon-ebssurrogate.source: total ---------------------------------------------------------------- 2537.60s

```

File: ebssurrogate/files/migrations/db/migrations/20221013153608_test.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- migrate:up
+
+
+-- migrate:down
+
```
Comment (sweatybridge) [inline ebssurrogate/files/migrations/db/migrations/20221013153608_test.sql]: can be deleted?

**Review by sweatybridge — APPROVED** [review]

File: ebssurrogate/files/migrations/db/migrations/20221013153608_test.sql (hunk 1 above)
Comment (Lakshmipathi) [reply ebssurrogate/files/migrations/db/migrations/20221013153608_test.sql]: ah, I didn't realise, let me go ahead and delete it.

**Review by pcnc — APPROVED** [review]

File: ansible/tasks/test-image.yml [hunk 2]
```diff
@@ -13,10 +27,21 @@
   failed_when: retval.rc != 0
   when: ebssurrogate_mode
 
+- name: Re-enable PG Sodium references in config
+  become: yes
+  become_user: postgres
+  shell:
+    cmd: mv /etc/postgresql/postgresql.conf.bak /etc/postgresql/postgresql.conf
```
Comment (darora) [inline ansible/tasks/test-image.yml:34]: seems out of sync with the use of sed to disable above?

File: ebssurrogate/files/migrations/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Migrations were pulled (in order) from:
+
+1. [init-scripts/postgres](https://github.com/supabase/infrastructure/tree/develop/init-scripts/postgres) => [db/init-scripts](db/init-scripts)
+2. [init-scripts/migrations](https://github.com/supabase/infrastructure/tree/develop/init-scripts/migrations) => [db/migrations](db/migrations)
+
+For compatibility with hosted projects, we include [migrate.sh](migrate.sh) that executes migrations in the same order as ami build:
+
+1. Run all `db/init-scripts` with `postgres` superuser role.
+2. Run all `db/migrations` with `supabase_admin` superuser role.
+3. Finalise role passwords with `/etc/postgres.schema.sql` if present.
```
Comment (darora) [inline ebssurrogate/files/migrations/README.md]: nit: `finalize`

**Review by darora — APPROVED** [review]

File: ansible/tasks/test-image.yml (hunk 2 above)
Comment (Lakshmipathi) [reply ansible/tasks/test-image.yml:34]: I had to temporarily remove sodium related entry `pgsodium.getkey_script` from the config file to start the process. So did `sed` to backup the original config and after running pgtap, now revert to the original config.  

Before this PR pgtap used to run against config from /var/lib/postgresql, now with this PR pgtap uses /etc/postgresql/postgresql.conf to run the tests.

**Pull request merged**